### PR TITLE
[wasm] benchmark sample modularized

### LINKED ISSUE #62357: [wasm] browser-bench appstart is broken on main
image
Seems like the startup changes probably broke it somehow. Ran into this while trying to use it to test something. cc @pavelsavara 
I tried to figure out why it seems to be retrying the config load operation constantly and from reading the code it shouldn't be happening, but it is. Not sure if that's related to the failure.

### PR DESCRIPTION
**Pull request by pavelsavara**
- Silence the errors just before we abort the iframe, because there all fetch promises in flight would be aborted inside the iframe and the errors would be printed
- Make benchmark more readable, separate the application in the iframe
- bind `console.log` and `console.error` as Kate suggested
- use `Module.printErr` to report all errors

Fixes https://github.com/dotnet/runtime/issues/62357

### CODE REVIEW

File: src/mono/wasm/runtime/exports.ts [hunk 1]
```diff
@@ -171,10 +171,10 @@ function initializeImportsAndExports(
     }
 
     if (!module.print) {
-        module.print = console.log;
+        module.print = console.log.bind(module);
     }
     if (!module.printErr) {
-        module.printErr = console.error;
+        module.printErr = console.error.bind(module);
```
Comment (kg) [inline src/mono/wasm/runtime/exports.ts:177]: it needs to bind ```console```, not anything else

**Review by kg — DISMISSED** [review]
Looks good overall but there are parts of this I'm not familiar with

File: src/mono/sample/wasm/browser-bench/frame-main.js [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+        },
+        onConfigLoaded: () => {
+            window.parent.resolveAppStartEvent("onConfigLoaded");
+            // Module.config.diagnostic_tracing = true;
+        },
+        onDotnetReady: async () => {
+            window.parent.resolveAppStartEvent("onDotnetReady");
+            try {
+                await frameApp.init({ BINDING });
+                window.parent.resolveAppStartEvent("done");
```
Comment (radekdoulik) [inline src/mono/sample/wasm/browser-bench/frame-main.js:36]:
```suggestion
```
I don't see use of this event and it is causing

    TypeError: Cannot read properties of null (reading 'resolveAppStartEvent')
        at Object.onDotnetReady (frame-main.js:36)

as the iframe is probably already unparented at this point.

**Review by radekdoulik — APPROVED** [review]
Besides the done event it LGTM.

**Pull request merged**